adapters/socketio_berkeley.c

```diff
@@ -644,6 +653,31 @@ static int set_target_network_interface(int socket, char* mac_address)
 }
 #endif //__APPLE__
 
+static void destroy_socket_io_instance(SOCKET_IO_INSTANCE* instance)
+{
+    if (instance->dns_resolver != NULL)
+    {
+        dns_resolver_destroy(instance->dns_resolver);
+    }
+
+    if (instance->hostname != NULL)
+    {
+        free(instance->hostname);
+    }
+
+    if (instance->target_mac_address != NULL)
```

Comment:
nit nit: i still don't think you need the checks but it's not a blocking issue.

Reply:
You are correct. From the c99 spec, "If ptr is a null pointer, no action occurs."
So there should be no check.